### PR TITLE
:sparkles: Add probe metadata about supported ecosystems

### PR DESCRIPTION
**Pull request by AdamKorcz**
#### What kind of change does this PR introduce?

feature

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

Adds metadata about probes' supported ecosystems in their `def.yml` files. Currently support covers languages and source code platforms.

- [x] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note

```

### CODE REVIEW

**Review by laurentsimon — APPROVED** [review]
LGTM. It's all private so we can change it if needed

File: probes/hasOpenSSFBadge/def.yml [hunk 1]
```diff
@@ -27,3 +27,9 @@ remediation:
     - Learn about best practices by following [the OpenSSF Best Practices Badge criteria](https://www.bestpractices.dev/en/criteria/0).
   markdown:
     - Learn about best practices by following [the OpenSSF Best Practices Badge criteria](https://www.bestpractices.dev/en/criteria/0).
+ecosystem:
+  languages:
+    - all
+  clients:
+    - cii_blob
+    - cii_http
```
Comment (spencerschrock) [inline probes/hasOpenSSFBadge/def.yml:35]: recommend dropping cii_blob and cii_http for reasons detailed above. Should be github and gitlab

**Review by spencerschrock — COMMENTED** [review]
just a quick comment on the clients

File: probes/fuzzedWithOSSFuzz/def.yml [hunk 2]
```diff
@@ -29,4 +29,16 @@ remediation:
...
+    - c
+    - c++
+    - go
+    - java
+    - javascript
+    - python
+    - rust
+    - typescript
+  clients:
+    - oss-fuzz
```
Comment (spencerschrock) [inline probes/fuzzedWithOSSFuzz/def.yml:44]: whoops missed one. I think this should be github/gitlab, not oss-fuzz. for reasons above

Comment (AdamKorcz) [reply probes/fuzzedWithOSSFuzz/def.yml]: https://github.com/ossf/scorecard/pull/3797/commits/7bcf9773f5e16ed35f16f40534b74af860cc72f5

File: probes/hasOpenSSFBadge/def.yml (hunk 1 above)
Comment (AdamKorcz) [reply probes/hasOpenSSFBadge/def.yml]: https://github.com/ossf/scorecard/pull/3797/commits/7bcf9773f5e16ed35f16f40534b74af860cc72f5

File: probes/hasOSVVulnerabilities/def.yml [hunk 3]
```diff
@@ -31,3 +31,11 @@ remediation:
...
     - If the vulnerability is in a dependency, update the dependency to a non-vulnerable version. If no update is available, consider whether to remove the dependency.
     - If you believe the vulnerability does not affect your project, the vulnerability can be ignored. To ignore, create an osv-scanner.toml ([example](https://github.com/google/osv.dev/blob/eb99b02ec8895fe5b87d1e76675ddad79a15f817/vulnfeeds/osv-scanner.toml)) file next to the dependency manifest (e.g. package-lock.json) and specify the ID to ignore and reason. Details on the structure of osv-scanner.toml can be found on [OSV-Scanner repository](https://github.com/google/osv-scanner#ignore-vulnerabilities-by-id).
+ecosystem:
+  languages:
+    - all
+  clients:
+    - github
+    - gitlab
+    - localdir
+    - osv
```
[the remark this answers is not in the document]
Comment (AdamKorcz) [reply probes/hasOSVVulnerabilities/def.yml]: https://github.com/ossf/scorecard/pull/3797/commits/7bcf9773f5e16ed35f16f40534b74af860cc72f5

File: finding/probe/probe.go [hunk 4]
```diff
@@ -57,11 +59,27 @@ type yamlRemediation struct {
...
+}
+
+var supportedClients = map[string]bool{
+	"github":   true,
+	"gitlab":   true,
+	"localdir": true,
+	"oss-fuzz": true,
+	"osv":      true,
+	"cii_blob": true,
+	"cii_http": true,
```
[the remark this answers is not in the document]
Comment (AdamKorcz) [reply finding/probe/probe.go]: https://github.com/ossf/scorecard/pull/3797/commits/7bcf9773f5e16ed35f16f40534b74af860cc72f5

**Comment by AdamKorcz** [comment]
@spencerschrock Ready for another review.

**Comment by spencerschrock** [comment]
> @spencerschrock Ready for another review.

The oss fuzz and cii best practices one dont have any clients, instead of being github and gitlab.

> Should be github and gitlab

**Comment by AdamKorcz** [comment]
> > @spencerschrock Ready for another review.
> 
> The oss fuzz and cii best practices one dont have any clients, instead of being github and gitlab.
> 
> > Should be github and gitlab

Added here: https://github.com/ossf/scorecard/pull/3797/commits/68086176d5b13ef27946e2d6db01eee47f218625

**Comment by spencerschrock** [comment]
Cool, maintainers can do another review during #2657

**Pull request merged**